docs/reference/runtime/Memory.md

```diff
@@ -1,51 +1,75 @@
+| User thread 1 stack |
+|---------------------|
+|          ^          |
+|          |          |
+|                     |
+|        Heap         |
++---------------------+
+|                     |
+| ZI: Global data     |
+|                     |
++---------------------+
+| ZI: Idle stack      |
++---------------------+
+| ZI: Timer stack     |
++---------------------+
+| ZI: Main stack      |
++---------------------+
+|                     |
+| ZI: Global data     |
+|                     |
++---------------------+
+| RW: Vector table    |
++=====================+   First address of RAM
+|                     |   Last address of flash
+|                     |
+|     Application     |
+|                     |
+|                     |
++---------------------+
+|                     |
+| Optional bootloader |
+|                     |
++---------------------+   First address of flash
 
-* The stack of the main thread (executing the main function).
-* The idle thread executed each time all the other threads are waiting for external or scheduled events. This is particularly useful for implementing energy saving strategies (like sleep).
-* The timer thread that executes all the time-scheduled tasks (periodic and nonperiodic).
-* The stack of OS scheduler itself (also used by the ISRs).
+```
 
-Stack checking is turned on for all threads, and the kernel will error if an overflow condition is detected.
+There are at least two kinds of memory in the system: flash and RAM.
 
-```
-+-------------------+   Last Address of RAM
-| Scheduler Stack   |
-+-------------------+
-|                   |   RAM
-|                   |
-|         ^         |
-|         |         |
-|    Heap Cont..    |
-+-------------------+
-| app thread n      |
-|-------------------|
-| app thread 2      |
-|-------------------|
-| app thread 1      |
-|-------------------|
-|         ^         |
-|         |         |
-|       Heap        |
-+-------------------+
-| ZI                |
-+-------------------+
-| ZI: OS drv stack  |
-+-------------------+
-| ZI: app thread 3  |
-+-------------------+
-| ZI: Idle Stack    |
-+-------------------+
-| ZI: Timer Stack   |
-+-------------------+
-| ZI: Main Stack    |
-+-------------------+
-| RW                |  
-+===================+   First Address of RAM
-|                   |
-|                   |   Flash
+### RAM
 
-```
+Inside RAM, we can distinguish two logical types: static and dynamic memory. The system uses each of them in different ways:
+* Static (zero initialized):
+  * Vector table.
+  * Global data.
+  * Static data.
+  * Stacks for default threads (main, timer, idle, scheduler/ISR).
+* Dynamic:
+  * Heap (dynamic data).
+  * Stacks for user threads. Mbed OS will dynamically allocate memory on heap for user thread's stacks.
+
+Stack checking is turned on for all threads, and the kernel errors if it detects an overflow condition.
+
+### Flash
+
+Flash is a read only memory (ROM) that contains:
+* Application code.
```

Comment:
As asked above, vector table might be here ?

Reply:
Fixed